atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresDdlTable.java

```diff
@@ -118,10 +126,67 @@ public void checkDatabaseVersion() {
 
     @Override
     public void compactInternally() {
+        try {
+            if (compactionSemaphore.tryAcquire()) {
+                checkLastCompactionTimeAndRunCompact();
+            }
+        } finally {
+            compactionSemaphore.release();
```

Comment:
This is wrong as it is always releasing semaphore, but may not have acquired it.